internal/controllers/gateway/tcproute_controller.go

```diff
@@ -349,6 +351,35 @@ func (r *TCPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, nil
 	}
 
+	// update "Programmed" condition if the TCPRoute is translated to Kong configuration.
+	// if the TCPRoute is not configured ad Kong side, leave it unchanged and requeue.
+	// if it is successfully configured, update its "Programmed" condition to True.
+	// if translation failures happens, update its "Programmed" condition to False.
+	debug(log, tcproute, "ensuring status contains Programmed condition")
+	if r.DataplaneClient.AreKubernetesObjectReportsEnabled() {
```

Comment:
Doesn't The programmed condition get updated to true in the TCPRoute controller? Something similar to the HTTPRoute version:

```go
statusUpdated, err := r.ensureParentsProgrammedCondition(ctx, httproute, gateways, metav1.ConditionTrue, ConditionReasonConfiguredInGateway, "")
		if err != nil {
			// don't proceed until the statuses can be updated appropriately
			debug(log, httproute, "failed to update programmed condition")
			return ctrl.Result{}, err
		}
		if statusUpdated {
			// if the status was updated it will trigger a follow-up reconciliation
			// so we don't need to do anything further here.
			debug(log, httproute, "programmed condition updated")
			return ctrl.Result{}, nil
		}
```

Reply:
thanks, fixed.